### PR TITLE
pr template: add docs changes optional link, misc enhancements

### PR DESCRIPTION
**Pull request by junr03**
Signed-off-by: Jose Nino <jnino@lyft.com>

pr template: add docs changes optional link, misc enhancements

*Description*: added an optional link to docs changes. Moved the optional fields after the required ones.

*Risk Level*: Low 

*Testing*: docs changes. No testing needed.

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -41,8 +35,17 @@ forms of testing, please use your best judgement or ask for guidance
...
+>Link to [Data Plane PR](https://github.com/envoyproxy/data-plane-api/pulls)]
+
+[Optional *Docs Changes*:]
+>Link to [Data Plane PR](https://github.com/envoyproxy/data-plane-api/pulls)]
+
+[Optional *Deprecated*:]
+>Description of what is deprecated.
 
-[optional *Release Notes*:] If this change is user impacting you **must** add a release note to
+[Optional *Release Notes*:]
```
Comment (mattklein123) [inline PULL_REQUEST_TEMPLATE.md]: Can you move this to the first optional thing to make sure people look at it?

Comment (htuch) [reply PULL_REQUEST_TEMPLATE.md]: I would be in favor of making this mandatory (enforced by a presubmit check) and having folks fill in `{n/a, na, n.a.}` etc. when there is no impact just to ensure that everyone gives this some thought.

Comment (mattklein123) [reply PULL_REQUEST_TEMPLATE.md]: I agree, but I don't think we should block this PR on that (we should update the text though to be required and have the user specify one of the options). I opened https://github.com/envoyproxy/envoy/issues/2133 to track automation.

Comment (junr03) [reply PULL_REQUEST_TEMPLATE.md]: I moved to mandatory section to force people to think about it.

Comment (mattklein123) [reply PULL_REQUEST_TEMPLATE.md]: I don't see any changes / DCO is broken. Can you fix?

Comment (junr03) [reply PULL_REQUEST_TEMPLATE.md]: I committed through the github UI, forgot about the DCO. Fixed now.

**Pull request merged**